handler/src/main/java/io/netty/handler/ssl/JdkSslClientContext.java

```diff
@@ -139,7 +141,35 @@ public JdkSslClientContext(
             Iterable<String> ciphers, CipherSuiteFilter cipherFilter, ApplicationProtocolConfig apn,
             long sessionCacheSize, long sessionTimeout) throws SSLException {
         this(certChainFile, trustManagerFactory, ciphers, cipherFilter,
-                toNegotiator(apn, false), sessionCacheSize, sessionTimeout);
+                toNegotiator(apn, false), sessionCacheSize, sessionTimeout, KeyStore.getDefaultType());
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param certChainFile an X.509 certificate chain file in PEM format.
+     *                      {@code null} to use the system default
+     * @param trustManagerFactory the {@link TrustManagerFactory} that provides the {@link TrustManager}s
+     *                            that verifies the certificates sent from servers.
+     *                            {@code null} to use the default.
+     * @param ciphers the cipher suites to enable, in the order of preference.
+     *                {@code null} to use the default cipher suites.
+     * @param cipherFilter a filter to apply over the supplied list of ciphers
+     * @param apn Application Protocol Negotiator object.
+     * @param sessionCacheSize the size of the cache used for storing SSL session objects.
+     *                         {@code 0} to use the default value.
+     * @param sessionTimeout the timeout for the cached SSL session objects, in seconds.
+     *                       {@code 0} to use the default value.
+     * @param keyStore the keyStore this context should use
+     * @deprecated use {@link SslContextBuilder}
+     */
+    @Deprecated
+    public JdkSslClientContext(
```

Comment:
I does not make sense to expose a public method and mark it as `@Deprecated`. I suspect this should be package-private